homeassistant/auth/__init__.py

```diff
@@ -257,7 +262,43 @@ def get_auth_mfa_module(self, module_id: str) \
         if not user.system_generated and client_id is None:
             raise ValueError('Client is required to generate a refresh token.')
 
-        return await self._store.async_create_refresh_token(user, client_id)
+        if token_type is None:
+            if user.system_generated:
+                token_type = models.TOKEN_TYPE_SYSTEM
+            else:
+                token_type = models.TOKEN_TYPE_NORMAL
+
+        if user.system_generated and token_type != models.TOKEN_TYPE_SYSTEM:
+            raise ValueError(
+                'System generated users can only have system type '
+                'refresh tokens')
+
+        if not user.system_generated and \
+                token_type == models.TOKEN_TYPE_SYSTEM:
+            raise ValueError(
+                'Only system generated users can have system type '
+                'refresh tokens')
+
+        if (token_type == models.TOKEN_TYPE_LONG_LIVED_ACCESS_TOKEN and
+                (client_id is None or client_id.lower().startswith(
+                    ('http://', 'https://')))):
+            raise ValueError('Client_id is not allowed start with http:// or '
+                             'https:// for long-lived access token')
+
+        if token_type == models.TOKEN_TYPE_LONG_LIVED_ACCESS_TOKEN:
```

Comment:
I think that this is a weird check. We shouldn't enforce this but instead leave this up to the user.

Reply:
Current is
1 client_id : 1 refresh_token : 1 long-lived access_token

Did you want to change to
1 client_id : m refresh_token : m long-lived access_token